src/app/client/src/app/modules/groups/components/group-form/group-form.component.ts

```diff
@@ -1,36 +1,251 @@
-    {name: 'Bengali Medium'}, {name: 'Punjabi Medium'}, {name: 'Urdu Medium'}];
-
-   this.classes = [  {name: 'Class 1'}, {name: 'Class 2'}, {name: 'Class 3'},
-    {name: 'Class 4'}, {name: 'Class 5'}, {name: 'Class 6'}, {name: 'Class 7'}, {name: 'Class 8'},
-    {name: 'Gujarat'}, {name: 'Harayana'}, {name: 'Himachal Pradesh'}, {name: 'Jammu And Kashmir'},
-    {name: 'Class 9'}, {name: 'Class 10'}];
+  constructor(private userService: UserService, private frameworkService: FrameworkService,
+    public resourceService: ResourceService, private toasterService: ToasterService, private channelService: ChannelService,
+    private orgDetailsService: OrgDetailsService, private fb: FormBuilder) { }
 
-   this.subjects = [  {name: 'English'}, {name: 'Hindi'}, {name: 'Tamil'},
-    {name: 'Malayalam'}, {name: 'Telugu'}, {name: 'Kannada Medium'},
-    {name: 'Bengali'}, {name: 'Punjabi'}, {name: 'Urdu'}];
+  ngOnInit() {
+    this.initializeForm();
+  }
+  private initializeForm() {
+    this.userService.userData$.subscribe(userdata => {
+      if (userdata && !userdata.err) {
+        this.userProfile = userdata.userProfile;
+      }
+    });
+    this.selectedOption = _.pickBy(_.cloneDeep(this.userProfile.framework), 'length') || {};
+    this.groupForm = this.fb.group({
+      groupName: ['', [ Validators.required ]],
+      board: [_.get(this.selectedOption, 'board') || [], [ Validators.required ]],
+      medium: [_.get(this.selectedOption, 'medium') || [], [ Validators.required ]],
+      gradeLevel: [_.get(this.selectedOption, 'gradeLevel') || [], [ Validators.required ]],
+      subject: [_.get(this.selectedOption, 'subject') || [], [ Validators.required ]],
+    });
+    this.editMode = _.some(this.selectedOption, 'length') || false ;
+    this.unsubscribe = this.isCustodianOrgUser().pipe(
+      mergeMap((custodianOrgUser: boolean) => {
+        this.custodianOrg = custodianOrgUser;
+        if (custodianOrgUser) {
+          return this.getFormOptionsForCustodianOrg();
+        } else {
+          return this.getFormOptionsForOnboardedUser();
+        }
+      }), first()).subscribe(data => {
+        this.formFieldOptions = data;
+        this.boardList = _.get(_.find(data, { 'code': 'board' }), 'range') || [];
+        this.mediumList = _.get(_.find(data, { 'code': 'medium' }), 'range') || [];
+        this.gradesList = _.get(_.find(data, { 'code': 'gradeLevel' }), 'range') || [];
+        this.subjectList = _.get(_.find(data, { 'code': 'subject' }), 'range') || [];
+      }, err => {
+        this.toasterService.warning(this.resourceService.messages.emsg.m0012);
+      });
+  }
+  private isCustodianOrgUser() {
+    return this.orgDetailsService.getCustodianOrg().pipe(map((custodianOrg) => {
+      if (_.get(this.userService, 'userProfile.rootOrg.rootOrgId') === _.get(custodianOrg, 'result.response.value')) {
+        return true;
+      }
+      return false;
+    }));
+  }
+  private getFormOptionsForCustodianOrg() {
```

Comment:
Move this method to service and keep component lean

Reply:
Moved some methods to service which has no interaction with form or component properties.